verdictd/app/src/PolicyEngine/opa/opaEngine.rs

```diff
@@ -0,0 +1,287 @@
+use serde_json::Value;
+use std::ffi::CStr;
+use std::fs::{remove_file, File};
+use std::io::prelude::*;
+use std::os::raw::c_char;
+use std::path::Path;
+use std::process::Command;
+
+/// Link import cgo function
+#[link(name = "opa")]
+extern "C" {
+    pub fn makeDecisionGo(policy: GoString, message: GoString) -> *mut c_char;
+}
+
+const POLICY_PATH: &str = "src/policyEngine/opa/policy/";
+
+/// String structure passed into cgo
+#[derive(Debug)]
+#[repr(C)]
+pub struct GoString {
+    pub p: *const c_char,
+    pub n: isize,
+}
+
+/// Introduce reference values, update or create new opa's policy files
+/// references (JSON)
+/// {
+///     "mrEnclave" : xxx
+///     "mrSigner" : xxx
+///     "productId" : xxx
+///     ...
+/// }
+pub fn set_reference(policy_name: &str, references: &str) -> bool {
+    // Deserialize the references in json format
+    let references: Value = match serde_json::from_str(references) {
+        Ok(res) => res,
+        Err(_) => {
+            println!("Json unmashall failed");
+            return false;
+        }
+    };
+
+    // Generate policy file from reference value
+    let mrEnclave: String = String::from("mrEnclave = ") + &references["mrEnclave"].to_string();
+    let mrSigner: String = String::from("mrSigner = ") + &references["mrSigner"].to_string();
+    let productId: String = String::from("productId = ") + &references["productId"].to_string();
+    let policy = "package demo\n\n".to_string()
```

Comment:
I think it's better to remove demo description.

Reply:
Do you mean to remove the "demo" information in "let policy = \"package demo\n\n\".to_string()"？